multigeiger/tube.cpp

```diff
@@ -35,22 +35,121 @@ TUBETYPE tubes[] = {
 
 volatile bool isr_GMC_cap_full;
 
+volatile unsigned long isr_hv_pulses;
+volatile bool isr_hv_charge_error;
+
 volatile unsigned int isr_GMC_counts;
 volatile unsigned long isr_count_timestamp;
 volatile unsigned long isr_count_time_between;
 
-static unsigned long hvpulse_timestamp;  // initialized via setup_tube -> gen_charge_pulses
+hw_timer_t * recharge_timer = NULL;
 
 // MUX (mutexes used for mutual exclusive access to isr variables)
 portMUX_TYPE mux_cap_full = portMUX_INITIALIZER_UNLOCKED;
 portMUX_TYPE mux_GMC_count = portMUX_INITIALIZER_UNLOCKED;
+portMUX_TYPE mux_hv = portMUX_INITIALIZER_UNLOCKED;
+
+void reschedule(hw_timer_t *timer, unsigned int cycles) {
+  timerAlarmWrite(timer, cycles, false);  // set alarm after <cycles>, do not repeat
+  timerWrite(timer, 0);
+  timerAlarmEnable(timer);
+}
+
+// How many HV capacitor charge pulses to generate for any later charge cycle.
+#define MAX_CHARGE_PULSES 333
+
+// How many HV capacitor charge pulses to generate for first charge.
+// After device power on, HV capacitor is completely empty.
+#define MAX_CHARGE_PULSES_INITIAL 3333
```

Comment:
I guess we could also simplify the code and use the higher value always.

The code will now stop doing recharges anyway if that amount of pulses does not fully charge the capacitor.

Reply:
As the charging stops once full anyway, a single value is enough IMO. 
Worst case during operation? So many pulse that the HV capacitor is empty again and must be recharged.
Lower value only required if we want to react more quickly with alert / warning during operation. Is that the case?

Reply:
Guess I'll just use one relatively high value.